odoo/models.py

```diff
@@ -4415,9 +4415,63 @@ def _load_records(self, data_list, update=False):
                 imd.browse(d_id).unlink()
                 to_create.append(data)
 
+        def must_update(fname, val, rec):
+            if fname not in rec._fields:
+                return True
+
+            field = rec._fields[fname]
+            if val is None or val is False:
+                return rec[fname] is not False
+
+            if field.type in ('many2many', 'one2many'):
+                if val == []:
+                    # nothing changed
+                    return False
+                if not isinstance(val, list):
+                    return True
+
+                if len(val) > 1:
+                    # multiple commands are complex to analyze, this optimization aims to
+                    # analyze only single-command updates which is fine since that's the
+                    # most common case in xml data
+                    return True
+
+                command, = val
+                # we want the linked records even if unactive
+                rec_val = rec.with_context(active_test=False)[field.name]
```

Comment:
Why do you compare the field's value from another context?  This looks inconsistent to me.

Reply:
It is actually inconsistent for `Command.CLEAR`, but not for `Command.LINK` and `Command.UNLINK`.